### PR TITLE
 Widgets/Button: New - Set gradient background to button

### LINKED ISSUE #3142: Feature Request: Button Gradient
It would be great to be able to have gradient backgrounds on buttons. (Reg and hover) This is pretty standard and web design people like to use gradients on buttons, especially for marketing purposes. Simple Gradient options like you already have for section backgrounds would be amazing! Thank you

### PR DESCRIPTION
**Pull request by StyleShit**
Closes #3142

### CODE REVIEW

File: includes/widgets/button.php [hunk 1]
```diff
@@ -314,16 +314,20 @@ protected function _register_controls() {
...
+			Group_Control_Background::get_type(),
 			[
-				'label' => __( 'Background Color', 'elementor' ),
-				'type' => Controls_Manager::COLOR,
-				'global' => [
-					'default' => Global_Colors::COLOR_ACCENT,
-				],
-				'selectors' => [
-					'{{WRAPPER}} .elementor-button' => 'background-color: {{VALUE}};',
+				'name' => 'background',
```
Comment (matipojo) [inline includes/widgets/button.php:320]: Did it keep backward compatibility?
Try to save a color with the old control, and then change it to the group control, did the old color still exist?

Comment (StyleShit) [reply includes/widgets/button.php:320]: Now it will

File: includes/widgets/button.php [hunk 2]
```diff
@@ -349,13 +356,20 @@ protected function _register_controls() {
...
-		$this->add_control(
-			'button_background_hover_color',
+		$this->add_group_control(
+			Group_Control_Background::get_type(),
 			[
-				'label' => __( 'Background Color', 'elementor' ),
-				'type' => Controls_Manager::COLOR,
-				'selectors' => [
-					'{{WRAPPER}} .elementor-button:hover, {{WRAPPER}} .elementor-button:focus' => 'background-color: {{VALUE}};',
+				'name' => 'background_hover',
```
Comment (matipojo) [inline includes/widgets/button.php:362]: Fix backward compatibility.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/widgets/button.php (hunk 2 above)
Comment (StyleShit) [reply includes/widgets/button.php]: Done

File: includes/widgets/button.php [hunk 3]
```diff
@@ -349,13 +356,23 @@ protected function _register_controls() {
...
+				'selector' => '{{WRAPPER}} .elementor-button:hover, {{WRAPPER}} .elementor-button:focus',
+				'fields_options' => [
+					'background' => [
+						'default' => 'classic',
+					],
+					'color' => [
+						'global' => [
+							'default' => Global_Colors::COLOR_SECONDARY,
+						],
+					],
```
Comment (matipojo) [inline includes/widgets/button.php:375]: Try remove

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/widgets/button.php (hunk 3 above)
Comment (matipojo) [reply includes/widgets/button.php:375]: @StyleShit any updates?

Comment (StyleShit) [reply includes/widgets/button.php]: Done

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**